internal/pkg/term/selector/local_selector.go

```diff
@@ -219,6 +281,45 @@ func (s *localFileSelector) listDockerfiles() ([]string, error) {
 	return dockerfiles, nil
 }
 
+// listDirsAndFiles returns the list of directories and files within the current
+// working directory and two subdirectory levels below.
+func (s *localFileSelector) listDirsAndFiles() ([]string, error) {
+	names, err := s.getDirAndFileNames(s.ws.ProjectRoot(), 3)
+	if err != nil {
+		return nil, err
+	}
+	return names, nil
+}
+
+// getDirAndFileNames recursively fetches directory and file names to the depth indicated. Hidden files and the copilot dir are excluded.
+func (s *localFileSelector) getDirAndFileNames(dir string, depth int) ([]string, error) {
+	wdDirsAndFiles, err := s.fs.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	var names []string
+	for _, file := range wdDirsAndFiles {
+		name := file.Name()
+		if strings.HasPrefix(name, ".") || name == "copilot" {
+			continue
+		}
+		relPathName := dir + "/" + name
+		wsRelPathName, err := s.ws.Rel(relPathName)
+		if err != nil {
+			return nil, fmt.Errorf("get path relative to workspace: %w", err)
+		}
+		names = append(names, wsRelPathName)
+		if depth > 0 && file.IsDir() {
+			subNames, err := s.getDirAndFileNames(relPathName, depth-1)
+			if err != nil {
+				return nil, fmt.Errorf("get dir and file names: %w", err)
```

Comment:
nit: I feel like we don't need to wrap in this case! Because we'd just get `get dir and file names: get dir and file names: get dir and file names: real err`
```suggestion
				return nil, err
```